doc/en/user/source/extensions/css/workshop/linestring.rst

```diff
@@ -0,0 +1,500 @@
+
+The parameters provided by SLD are general purpose and should be compatible with any rendering engine.
+
+To take greater control over the GeoServer rendering engine we can use "vendor specific" parameters. These hints are used specifically for the GeoServer rendering engine and will be ignored by other systems. The GeoServer rendering engine marks each vendor specific parameter with the prefix **-gt-**.
+
+#. The ability to take control of the labeling process is exactly the kind of hint a vendor specific parameter is intended for.
+    
+   Update ``line_example`` with the following:
+
+   .. code-block:: css
+
+      * {
+        stroke: blue;
+        label: [name];
+        font-fill: black;
+        label-offset: 7px;
+        -gt-label-padding: 10;
+      }
+
+#. The parameter **-gt-label-padding** provides additional space around our label for use in collision avoidance.
+
+   .. code-block:: css
+      :emphasize-lines: 6
+   
+      * {
+        stroke: blue;
+        label: [name];
+        font-fill: black;
+        label-offset: 7px;
+        -gt-label-padding: 10;
+      }
+
+#. Each label is now separated from its neighbor, improving legibility.
+
+   .. image:: img/line_label_3.png
+
+Scale
+-----
+
+This section explores the use of attribute selectors and the :kbd:`@scale` selector together to simplify the road dataset for display.
+
+#. Replace the `line_example` CSS definition with:
+
+   .. code-block:: css
+
+      [scalerank < 4] {
+        stroke: black;
+      }
+
+#. And use the :guilabel:`Map` tab to preview the result.
+
+   .. image:: img/line_04_scalerank.png
+
+#. The **scalerank** attribute is provided by the Natural Earth dataset to allow control of the level of detail based on scale. Our selector short-listed all content with scalerank 4 or lower, providing a nice quick preview when we are zoomed out.
+
+#. In addition to testing feature attributes, selectors can also be used to check the state of the rendering engine.
+
+   Replace your CSS with the following:
+
+   .. code-block:: css
+
+      [@scale > 35000000] {
+         stroke: black;
+      }
+      [@scale < 35000000] {
+         stroke: blue;
+      }
+
+#. As you adjust the scale in the :guilabel:`Map` preview (using the mouse scroll wheel) the color will change between black and blue. You can read the current scale in the bottom right corner, and the legend will change to reflect the current style.
+
+   .. image:: img/line_05_scale.png
+
+#. Putting these two ideas together allows control of level detail based on scale:
+
+   .. code-block:: css
+
+      [@scale < 9000000] [scalerank > 7] {
```

Comment:
To be correct, I think that ``[scalerank > 7]`` should just be taken out (like in the example later on in the points section). Also, this whole example should incorporate ``>=`` into its clauses instead of just ``>``.